app/models/question.rb

```diff
@@ -36,6 +37,15 @@ def created_by?(user)
     self.user.id == user.id
   end
 
+  def has_approved_answer?
+    !self.approved_answer_id.nil?
+  end
+
+  def unapproved_answers
+    id_to_exclude = self.approved_answer_id
+    self.answers.where{id.not_eq id_to_exclude}
```

Comment:
Redundant `self` detected.<br>Space missing to the left of {.<br>Space missing inside {.<br>Space missing inside }.